okhttp/src/test/java/okhttp3/CallHandshakeTest.kt

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (C) 2020 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3
+
+import mockwebserver3.MockResponse
+import mockwebserver3.MockWebServer
+import okhttp3.CipherSuite.Companion.TLS_AES_128_GCM_SHA256
+import okhttp3.CipherSuite.Companion.TLS_AES_256_GCM_SHA384
+import okhttp3.CipherSuite.Companion.TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384
+import okhttp3.CipherSuite.Companion.TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
+import okhttp3.testing.PlatformRule
+import okhttp3.tls.internal.TlsUtil.localhost
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.RegisterExtension
+import javax.net.ssl.SSLSocket
+
+class CallHandshakeTest {
+  private lateinit var client: OkHttpClient
+  private lateinit var server: MockWebServer
+  val handshakeCertificates = localhost()
+
+  @RegisterExtension
+  @JvmField
+  val clientTestRule = OkHttpClientTestRule()
+
+  @RegisterExtension
+  @JvmField
+  var platform = PlatformRule()
+
+  private lateinit var handshakeEnabledCipherSuites: List<String>
+  private lateinit var defaultEnabledCipherSuites: List<String>
+  private lateinit var defaultSupportedCipherSuites: List<String>
+
+  val expectedModernTls12CipherSuites = listOf(TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384, TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384)
```

Comment:
why is this list so short?

Reply:
Given a fixed set of platforms we test against in CI, there are only a certain number of likely results for the first cipher.  But at least if we curate any valid high strength choices here, it's a smoke test if it ever breaks and starts using a weaker one.